docs/source/basics/tokenizers.rst

```diff
@@ -0,0 +1,320 @@
+.. _tokenizers_usage_label:
+
+==========
+Tokenizers
+==========
+
+Tokenizers are a key component of any LLM. They convert raw text into token IDs, which index into embedding vectors that are
+understood by the model.
+
+In torchtune, tokenizers play the role of converting Messages object into token IDs and any necessary model-specific special tokens.
+They use byte-pair encoding algorithms to convert raw strings into integer IDs. These are usually based on SentencePiece or TikToken.
+
+Downloading tokenizers from Hugging Face
+----------------------------------------
+
+Use the ``tune download`` command to download tokenizers from Hugging Face. They are found on the model page and are downloaded with
+the model weights.
+
+.. code-block:: bash
+
+    tune download mistralai/Mistral-7B-v0.1 --output-dir /tmp/Mistral-7B-v0.1 --hf-token <HF_TOKEN>
+    cd /tmp/Mistral-7B-v0.1/
+    ls tokenizer.model
+    # tokenizer.model
+
+Loading tokenizers from file
+----------------------------
+
+Once you've downloaded the tokenizer file, you can load it into the corresponding tokenizer class by pointing
+to the file path of the tokenizer model in your config or in the constructor. You can also pass in a custom file path if you've already
+downloaded it to a different location.
+
+.. code-block:: python
+
+    from torchtune.models.mistral import mistral_tokenizer
+
+    m_tokenizer = mistral_tokenizer("/tmp/Mistral-7B-v0.1/tokenizer.model")
+    type(m_tokenizer)
+    # <class 'torchtune.models.mistral._tokenizer.MistralTokenizer'>
+
+.. code-block:: yaml
+
+    tokenizer:
+      _component_: torchtune.models.mistral.mistral_tokenizer
+      path: /tmp/Mistral-7B-v0.1/tokenizer.model
+
+Setting max sequence length
+---------------------------
+
+.. code-block:: python
+
+    from torchtune.models.mistral import mistral_tokenizer
+
+    m_tokenizer = mistral_tokenizer("/tmp/Mistral-7B-v0.1/tokenizer.model", max_seq_len=8192)
+
+.. code-block:: yaml
+
+    tokenizer:
+      _component_: torchtune.models.mistral.mistral_tokenizer
+      path: /tmp/Mistral-7B-v0.1/tokenizer.model
+      max_seq_len: 8192
+
+
+Prompt templates
+----------------
+
+For more details on what prompt templates are and when you should use them, see :ref:`prompt_templates_usage_label`. Prompt templates
+are passed into the tokenizer and will be automatically applied for the dataset you are fine-tuning on. You can pass it in two ways:
+- A string dotpath to a prompt template class, i.e., "torchtune.models.mistral.MistralChatTemplate" or "path.to.my.CustomPromptTemplate"
+- A dictionary that maps role to a tuple of strings indicating the text to add before and after the message content
+
+Defining via dotpath string
```

Comment:
These are all really helpful examples and I wonder if it's worth shifting these into the prompt template docs and explain a bit how they're used/consumed by tokenizers?